src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -372,14 +372,9 @@ function MoneyRequestConfirmationList({
     // Calculate and set tax amount in transaction draft
     useEffect(() => {
         const taxAmount = getTaxAmount(transaction, policy).toString();
-        const amountInSmallestCurrencyUnits = CurrencyUtils.convertToBackendAmount(Number.parseFloat(taxAmount));
-
-        if (transaction?.taxAmount && previousTransactionAmount === transaction?.amount && previousTransactionCurrency === transaction?.currency) {
-            return IOU.setMoneyRequestTaxAmount(transactionID, transaction?.taxAmount ?? 0, true);
-        }
```

Comment:
I mostly think this was pre optimization. When the user updates transactionAmount, previousTransactionAmount was not being changed, and we we were not setting updated taxAmount in Onyx

Reply:
No, this introduced a bug where taxAmount is not being updated in draft transaction. looking...

Reply:
This is fixed now with latest commit - https://github.com/Expensify/App/pull/40240/commits/c671c7a080a35e331fbfc6476895e64bb2aebc28